resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -381,4 +382,32 @@ object KubernetesUtils extends Logging {
       }
     }
   }
+
+  @Since("3.3.0")
+  def loadFeatureStep(conf: KubernetesConf, className: String): KubernetesFeatureConfigStep = {
+    val constructors = Utils.classForName(className).getConstructors
```

Comment:
I was thinking about how we could avoid reflections here and my idea is the following:
1) Extending  `KubernetesFeatureConfigStep` with a new method `def init(config: KubernetesConf) {}` 
2) Introducing two new traits (`KubernetesDriverFeatureConfigStep`/`KubernetesExecutorFeatureConfigStep`) where the `init` accepts `KubernetesDriverConf` and `KubernetesExecutorConf` respectively. The `init` in the derived classes this way is not overriding the init of `KubernetesFeatureConfigStep`. So we cannot create a common `loadFeatureStep` where `KubernetesConf` is passed but it can be a function with generic parameter.

And finally we can extend the doc of `KUBERNETES_DRIVER_POD_FEATURE_STEPS` and `KUBERNETES_EXECUTOR_POD_FEATURE_STEPS` to mention those two specific traits.

This way the custom feature step's API will contain a clue that configs can be used to initialize. 

WDYT?

Reply:
Thanks for your suggestion.

> avoid reflections here and my idea is the following

Yes, it's a little bit complex and some hack taste, but the advantage would be same experience and interface with native feature step, and init way is simple, but perhaps bring some different interface for user.

If I understand correctly your idea, this https://github.com/apache/spark/pull/35345 migth be the alternative way you mentioend. Could you take a look on it? @attilapiros (I haven't updated the doc so you could just review on code implementation. )

Reply:
Today I was busy with some internal stuffs so I just had time to run through it and although it looks better I will have some comments but only on Monday...

Reply:
Much thanks for your help!